Tests/BackendIntegrationTests/StoreKitIntegrationTests.swift

```diff
@@ -433,10 +453,97 @@ class StoreKit1IntegrationTests: BaseBackendIntegrationTests {
 
 }
 
-private extension StoreKit1IntegrationTests {
+// MARK: - Observer Mode tests
+
+class BaseStoreKitObserverModeIntegrationTests: BaseStoreKitIntegrationTests {
+
+    override class var observerMode: Bool { return true }
+
+}
+
+class StoreKit2ObserverModeIntegrationTests: StoreKit1ObserverModeIntegrationTests {
+
+    override class var storeKit2Setting: StoreKit2Setting { return .enabledForCompatibleDevices }
+
+    @available(iOS 15.0, tvOS 15.0, watchOS 8.0, macOS 12.0, *)
+    func testPurchaseInDevicePostsReceipt() async throws {
+        try AvailabilityChecks.iOS15APIAvailableOrSkipTest()
+
+        try await self.purchaseProductFromStoreKit()
+
+        let info = try await Purchases.shared.restorePurchases()
```

Comment:
I'm confused about this - when we do restore we're forcing a receipt post, so this effectively is just testing that if the purchase there it gets posted, right? 
like, shouldn't this test just be "purchase from storekit, then verify entitlement went through"? Same question for the other one

Reply:
To be clear, my concern is that the minute that we call `restorePurchases`, we're not really testing observer mode behavior

Reply:
Okay I see what you're saying. I guess it's a very basic test. What else would you do if we don't restore purchases to verify what the SDK does in this case of purchasing from "outside" the SDK?

Reply:
after a purchase was made from "outside", the listeners should get notified and we should: 
- get a notification from the customerInfoChanged delegate
- if you call getCustomerInfo, it should have the entitlements
If those are not happening, then observer mode isn't working. 
It could be that it just doesn't work well with StoreKitTest or something, but if we call restorePurchases we do a force sync to backend, so we're not really doing what someone in observer mode would be doing

Reply:
Oh cool, let me try changing the test to that then.

Reply:
I updated this using #2134.

The good news: this works great! It waits for the transaction to be detected by the listener, which posts it to the backend, and then we see the entitlement.
The bad news: for some reason the transaction listener doesn't get the update until like 20-30 seconds later, which makes this test extremely slow (therefore why I added such a high `timeout`).

I think it's a very super useful test, but it being so slow is annoying. What do you think?